src/schedule/task.c

```diff
@@ -56,9 +56,13 @@ enum task_state task_main_primary_core(void *data)
 		/* sleep until next IPC or DMA */
 		wait_for_interrupt(0);
 
+#ifdef __ZEPHYR__
+		if (!ipc->task_mask & IPC_TASK_POWERDOWN)
```

Comment:
I think I wrote that there was more than one precedence bug to fix, didn't I

Reply:
Yeah thanks, did not catch this one sorry 👍

Reply:
I think it's all of them now.